src/Azure.Functions.Cli/Kubernetes/KubernetesHelpers.cs

```diff
@@ -146,18 +159,28 @@ internal static string GetKedaResources(string @namespace)
             int? cooldownPeriod = null,
             string serviceType = "LoadBalancer",
             int? minReplicas = null,
-            int? maxReplicas = null)
+            int? maxReplicas = null,
+            string keysSecretCollectionName = null,
+            bool mountKeysAsContainerVolume = false)
         {
             ScaledObjectV1Alpha1 scaledobject = null;
             var result = new List<IKubernetesResource>();
             var deployments = new List<DeploymentV1Apps>();
             var httpFunctions = triggers.FunctionsJson
                 .Where(b => b.Value["bindings"]?.Any(e => e?["type"].ToString().IndexOf("httpTrigger", StringComparison.OrdinalIgnoreCase) != -1) == true);
             var nonHttpFunctions = triggers.FunctionsJson.Where(f => httpFunctions.All(h => h.Key != f.Key));
+            keysSecretCollectionName = string.IsNullOrEmpty(keysSecretCollectionName) ? string.Concat($"func-keys-kube-secret-{name}") : keysSecretCollectionName;
             if (httpFunctions.Any())
             {
                 int position = 0;
                 var enabledFunctions = httpFunctions.ToDictionary(k => $"AzureFunctionsJobHost__functions__{position++}", v => v.Key);
+                //Environment variables for the func app keys
+                var funcKeyEnvironmentVariables = FuncAppKeysHelper.FuncKeysKubernetesEnvironVariables(keysSecretCollectionName, mountKeysAsContainerVolume);
+                foreach (var environmentVar in funcKeyEnvironmentVariables)
+                {
+                    enabledFunctions.TryAdd(environmentVar.Key, environmentVar.Value);
```

Comment:
variables should describe their content. Create a new dictionary that merges the 2 collections with a proper name